tests/bdd/features/rebuild/test_log_based_rebuild.py

```diff
@@ -194,75 +192,54 @@ def a_child_becomes_faulted():
 @when("a non-local child becomes faulted")
 def a_non_local_child_becomes_faulted():
     """a non-local child becomes faulted."""
-    pass
-
-
-@when(
-    "network connection between nexus and a replica node temporarily goes down and restored"
-)
-def _():
-    """network connection between nexus and a replica node temporarily goes down and restored."""
-    pass
+    a_child_becomes_faulted()
 
 
 @when("the replica is not online again within the timed-wait period")
 def the_replica_is_not_online_again_within_the_timed_wait_period():
     """the replica is not online again within the timed-wait period."""
     # The replica shouldn't be available on this node still.
-    time.sleep(2)
+    time.sleep(WAIT_FOR_FULL_REBUILD)
     replica = ApiClient.replicas_api().get_node_replicas(NODE_2_NAME)
     assert len(replica) == 0, "Replica not expected to be available on " + NODE_2_NAME
 
 
 @when("the replica is online again within the timed-wait period")
 def the_replica_is_online_again_within_the_timed_wait_period():
     """the replica is online again within the timed-wait period."""
-    # The control plane should have attempted the online child by now.
-    # Let's wait a few moments.
-    time.sleep(2)
+    # this sleep would ensure some new writes to be in the log
+    sleep(SLEEP_BEFORE_START)
+    Docker.restart_container(NODE_2_NAME)
+    # allow replica to be online after restart
+    check_replica_online()
+
+
+@retry(wait_fixed=500, stop_max_attempt_number=20)
+def check_replica_online():
     replica = ApiClient.replicas_api().get_node_replicas(NODE_2_NAME)
     if len(replica) == 1:
         assert replica[0].state == ReplicaState("Online")
     else:
-        # assert False, "Expected a replica to be available again on " + NODE_2_NAME
-        pass
-
-
-@when("the volume target moves to a different node before the log-based rebuild starts")
-def _():
-    """the volume target moves to a different node before the log-based rebuild starts."""
-    pass
+        assert False, "Expected a replica to be available again on " + NODE_2_NAME
 
 
 @then("a full rebuild starts after some time")
+@retry(wait_fixed=500, stop_max_attempt_number=20)
 def a_full_rebuild_starts_after_some_time():
     """a full rebuild starts after some time."""
     vol = ApiClient.volumes_api().get_volume(VOLUME_UUID)
     target = vol.state.target
     childlist = target["children"]
     assert len(childlist) == NUM_VOLUME_REPLICAS
+    check_child_removal()
     for child in childlist:
-        if child["state"] == ChildState("Faulted") or child["state"] == ChildState(
-            "Degraded"
-        ):
+        if child["state"] == ChildState("Degraded"):
```

Comment:
Should we check  "Degraded OR Online" ? If by chance the rebuild has finished, the state is going to be Online.

Reply:
I did check rebuild history for partial rebuild scenario. With current wait of 5 secs before container start and randwrite we get 24 seconds of rebuild period. This should be Okay and if we are checking online it will be true for already healthy child also right?

Reply:
hmm right, we don't want to check untouched online children here. Another way could be to check like
 `Degraded OR (Online && child["rebuild_progress"] != 0)`
 But if with current timing based runs this is consistently OK then perhaps fine, but is a bit flaky.